pkg/manager/manager.go

```diff
@@ -641,7 +650,7 @@ func setOptionsDefaults(options Options) Options {
 
 	// Allow newRecorderProvider to be mocked
 	if options.newRecorderProvider == nil {
-		options.newRecorderProvider = intrec.NewProvider
+		options.newRecorderProvider = newProvider
```

Comment:
To be honest. Not sure if this is the right way to solve this problem.

With this PR, none of the implementations of the `newRecorderProvider` will use the handed over `makeBroadcaster` parameter.
This means as far as I can tell that options.makeBroadcaster is never used now (for the manager)

If that is in fact what we want to do, why not drop the makeBroadcaster parameter from newRecorderProvider and the makeBroadcaster field from the options and do this

```go
	// Allow newRecorderProvider to be mocked
	if options.newRecorderProvider == nil {
		options.newRecorderProvider = func(config *rest.Config, httpClient *http.Client, scheme *runtime.Scheme, logger logr.Logger) (*intrec.Provider, error) {
			return intrec.NewProvider(config, httpClient, scheme, logger, func() (record.EventBroadcaster, bool) {
				return record.NewBroadcaster(), true
			})
		}
	}
```

Is that actually the new behavior we want? My implementation is equivalent to the current implementation just simpler.

It seems the effect is basically that the manager is not using the passed in EventBroadcaster anymore?

Reply:
Do we know why the current implementation produces duplicate events?

Reply:
Trying to summarize (https://github.com/kubernetes-sigs/controller-runtime/issues/1576#issuecomment-871987952)

In the case that an EventBroadcaster is passed in the following happens:
* we create 1 recorderProvider with this EventBroadCaster in manager
* we create 1 recorderProvider with this EventBroadCaster in cluster
* The result is that on the passed in EventBroadCaster we end up with duplicate watches
* Because of this every event will be sent twice

The current solution in this PR is to just use a separate broadcaster (record.NewBroadcaster()) in manager instead of the passed in broadcaster.

I have some doubts if just ignoring the passed in EventBroadcaster for the manager is the right way to solve this.

Wouldn't it be better to ensure we just have one recorderProvider that we use in both the manager and the cluster?



Side note: let's imagine we implement some sort of multi cluster support and then we have more than one cluster. I think we probably would get the same issue again.

Reply:
> Trying to summarize ([#1576 (comment)](https://github.com/kubernetes-sigs/controller-runtime/issues/1576#issuecomment-871987952))
> 
> In the case that an EventBroadcaster is passed in the following happens:
> 
> * we create 1 recorderProvider with this EventBroadcaster in manager
> * we create 1 recorderProvider with this EventBroadcaster in cluster
> * The result is that on the passed in EventBroadcaster we end up with duplicate watches
> * Because of this every event will be sent twice
> 
> The current solution in this PR is to just use a separate broadcaster (record.NewBroadcaster()) in manager instead of the passed in broadcaster.
> 
> I have some doubts if just ignoring the passed in EventBroadcaster for the manager is the right way to solve this.
> 
> Wouldn't it be better to ensure we just have one recorderProvider that we use in both the manager and the cluster?
> 
> Side note: let's imagine we implement some sort of multi cluster support and then we have more than one cluster. I think we probably would get the same issue again.

i want to set event qps/burst,so need EventBroadcaster options.